### PR TITLE
test(form): playwright refactor

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

Refactor Form tests from Cypress to Playwright

### Current behaviour

Tests are currently Cypress

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions
- [ ] Run Form tests in Playwright Test Runner to check if the `fieldset.pw.tsx` file passed
- [ ] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/form/form.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+  test(`should pass tests for WithButtonsAlignedToTheLeft example`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<WithButtonsAlignedToTheLeft />);
+
+    await checkAccessibility(page);
+  });
+
+  test(`should pass tests for WithCustomFooterPadding example`, async ({
```
Comment (sianford) [inline src/components/form/form.pw.tsx:334]: This should click on the `Open Preview` button first as currently it's just testing the default frame with the `Open Preview` button and not the Form with the custom footer padding.

File: src/components/form/form.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+    mount,
+    page,
+  }) => {
+    await mount(<WithCustomFooterPadding />);
+
+    await checkAccessibility(page);
+  });
+
+  test(`should pass tests for WithDifferentSpacing example`, async ({
```
Comment (sianford) [inline src/components/form/form.pw.tsx:343]: Is this test needed? It's not really proving anything from an accessibility perspective.

File: src/components/form/form.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+    page,
+  }) => {
+    await mount(<InDialogWithStickyFooter />);
+
+    const dialogButton = dataComponentButtonByText(page, "Open Preview");
+    await dialogButton.click();
+    await checkAccessibility(page);
+  });
+
+  test(`should pass tests for OverrideFieldSpacing example`, async ({
```
Comment (sianford) [inline src/components/form/form.pw.tsx:298]: I don't think this needs an accessibility test. Having spacing overridden would not cause a violation.

**Review by DipperTheDan — COMMENTED** [review]
Nothing more to add for me. Just address the comments left by @sianford and I'll approve ✅

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**